### PR TITLE
[Feature] Support Schema in MaxCompute Source

### PR DESCRIPTION
**Pull request by ruanwenjun**
## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by Hisoka-X** [comment]
Can you add new test case for this new feature? Like create MaxCompute Source with schema config and assert it.

**Comment by ruanwenjun** [comment]
> Can you add new test case for this new feature? Like create MaxCompute Source with schema config and assert it.

OK, I add MaxcomputeSourceTest for this, will create MaxComputeSource and use prepare method to initialize the typeInfo

**Review by hailin0 — APPROVED** [review]
LGTM

File: seatunnel-connectors-v2/connector-maxcompute/src/main/java/org/apache/seatunnel/connectors/seatunnel/maxcompute/catalog/MaxComputeCatalog.java [hunk 1]
```diff
@@ -0,0 +1,160 @@
+/*
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-maxcompute/src/main/java/org/apache/seatunnel/connectors/seatunnel/maxcompute/catalog/MaxComputeCatalog.java:1]: Shall we use test case to cover this?

**Review by Hisoka-X — APPROVED** [review]

**Pull request merged**